config/initializers/cors.rb

```diff
@@ -0,0 +1,16 @@
+# Be sure to restart your server when you modify this file.
+
+# Avoid CORS issues when API is called from the frontend app.
+# Handle Cross-Origin Resource Sharing (CORS) in order to accept cross-origin Ajax requests.
+
+# Read more: https://github.com/cyu/rack-cors
+
+# Rails.application.config.middleware.insert_before 0, Rack::Cors do
+#   allow do
+#     origins "example.com"
+#
+#     resource "*",
+#       headers: :any,
+#       methods: [:get, :post, :put, :patch, :delete, :options, :head]
+#   end
+# end
```

Comment:
Hi, I noticed that you did not make any changes to your code regarding the cors issue. This is a very important step to ensure that your application can communicate with the Rails API without any errors. Here is how you can enable cors in your Rails backend:

- Add the `rack-cors` gem to your Gemfile and run `bundle install`
- In your `config/initializers/cors.rb` file, uncomment the following code and replace `example.com` with your frontend domain:

```ruby
Rails.application.config.middleware.insert_before 0, Rack::Cors do
  allow do
    origins 'example.com'

    resource '*',
      headers: :any,
      methods: [:get, :post, :put, :patch, :delete, :options, :head]
  end
end
```

- Restart your Rails server and test your application again

Please make this change as soon as possible and let me know if you have any questions or issues. Thank you. 😊

Source: Conversation with Bing, 2/3/2024
(1) github.com. https://github.com/dgcfung/react_on_rails/tree/38635b021a870d11157795672690d90fc57f01f3/README.md.